【バグ】 macOS + clang 14.0.0 doesn't recognize std::ranges::transform in perception/identification.cpp

## 環境

- OS: macOS 12.6.2 with Apple clang 14.0.0 (clang-1400.0.29.202) from Xcode 14.2 (14C18)
- バージョン: 3.0.0Alpha72 + changes up to https://github.com/hengband/hengband/commit/b6348f09e70ebbfc13890b33e5eb1b8ecdab35e7

## バグの詳細

Compiling the develop branch on macOS 12.6.2 with Apple clang 14.0.0 results in a compiler error in perception/identification.cpp: `perception/identification.cpp:50:18: error: no member named 'transform' in namespace 'std::ranges'`.  The command line that make prints out for compiling perception/identification.cpp is  `./gcc-wrap g++ -std=gnu++20 -DHAVE_CONFIG_H -I. -I../src       -g -O2 -MT perception/identification.o -MD -MP -MF $depbase.Tpo -c -o perception/identification.o perception/identification.cpp && mv -f $depbase.Tpo $depbase.Po` .

## 再現手順

1. Clone the git repository, set the branch to develop, and update the submodule.
2. Configure the build with `./bootstrap; ./configure --disable-japanese --disable-pch --disable-worldscore`
3. Run `make` to build.

## 期待される正常な動作

The compilation runs to completion without an error.

## 補足事項

Looking at the header file for std::ranges in /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1 , I don't see transform there or in the header files in __ranges/*.h that the ranges header pulls in.  There is a transform_view in std::ranges.  Is there a way to write what std::ranges::transform is supposed to do with std::transform?

--- a/src/perception/identification.cpp
+++ b/src/perception/identification.cpp
@@ -47,7 +47,7 @@ bool screen_object(PlayerType *player_ptr, ItemEntity *o_ptr, BIT_FLAGS mode)
     const auto item_text_lines = shape_buffer(item_text, 77 - 15);
 
     int i = 0;
-    std::ranges::transform(item_text_lines, &info[i], [](const auto &line) { return line.data(); });
+    std::transform(item_text_lines.begin(), item_text_lines.end(), &info[i], [](const auto &line) { return line.data(); });
     i += item_text_lines.size();
 
     if (o_ptr->is_equipment()) {

